### PR TITLE
fix(#8790): fix api/deploy-info version for tags

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

`api/deploy-info` returns valid semver version for branches (like: `4.5.1-branch-name.4324242`) but sends an invalid semver version for tags (like `4.5.1.4327432`). 
Historically, the version for branches was the branch name, and it gets used in multiple places (in admin app, in the staging ddoc name. etc) so it's much more complicated to "standardize" the version we set in the staging ddoc. 

This change is limited to forcing the response of `api/deploy-info.version` to be semver valid. 

#8790

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:valid-semver/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:valid-semver/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:valid-semver/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/src/services/monitoring.js [hunk 1]
```diff
@@ -23,24 +24,22 @@ const fromEntries = (keys, value) => {
...
   if (seq) {
     const parts = seq.split('-');
-    if (parts.length) {
-      const result = parseInt(parts[0]);
-      if (!isNaN(result)) {
-        return result;
-      }
+    const result = parseInt(parts?.[0]);
+    if (!isNaN(result)) {
+      return result;
```
Comment (dianabarsan) [inline api/src/services/monitoring.js:29]: Sonar complaint for complexity.

File: tests/e2e/upgrade/upgrade.wdio-spec.js [hunk 2]
```diff
@@ -70,6 +71,12 @@ describe('Performing an upgrade', () => {
     });
   });
 
+  // TODO Enable this test after 4.6.0 is released
```
Comment (latin-panda) [inline tests/e2e/upgrade/upgrade.wdio-spec.js:74]: If not yet, please create a ticket in milestone 4.7

**Review by latin-panda — APPROVED** [review]
Nice improvements! It's good!

File: tests/e2e/upgrade/upgrade.wdio-spec.js (hunk 2 above)
Comment (dianabarsan) [reply tests/e2e/upgrade/upgrade.wdio-spec.js:74]: I was planning to create the ticket after this gets merged

**Pull request merged**